server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -696,6 +696,10 @@ private static void failIfMatchesRoutingPath(DocumentParserContext context, Stri
      */
     private static void parseCopyFields(DocumentParserContext context, List<String> copyToFields) throws IOException {
         for (String field : copyToFields) {
+            if (context.mappingLookup().inferenceFields().get(field) != null) {
```

Comment:
Is this even something we should be checking at runtime? Shouldn't we just remove copy_to that targets inference fields from the mapping that the parser works with up-front so we save this cost?

Reply:
I see your point. Maybe changing the actual field mapping would be too much - how about adding a new structure on the field type lookup for dealing with the copy_to list of fields to parse?

I've done a draft on [this PR](https://github.com/carlosdelest/elasticsearch/pull/14), WDYT?

Reply:
Is it really about cost? I mean it should be transparent performance wise. I don't think we should add a new structure in the lookup just for this.

Reply:
Yea lets not add a new structure, anything that increases the size of these things makes me uneasy after many-shards :) I was just thinking, if we know all the fields we don't want to copy_to upfront, why not just filter the `copyToFieldsList` at parser build time? Is that really something that is hard to do?

But on second thought! IS this even ok? Do we even want to allow this kind of mapping conflict? Shouldn't we just not have a copy_to definition that targets a field that gets populated by this mapper to begin with?

Reply:
We want to allow copy_to to target semantic text fields and we need to access this information somewhere so that we can do the actual copy in the coordinated node (when the inference is applied). The only difference here is on where we apply the copy_to. For semantic text fields it happens on the coordinating node, for other fields it happens during the parsing on the shard.

Reply:
> why not just filter the copyToFieldsList at parser build time? Is that really something that is hard to do?

Do you mean to use the DocumentParserContext.copyToFields() instead of the copyTo() information in the mapping? AFAIU this list is used for a different purpose.

If we're not using a new structure in field type lookup, I guess the two available options would be:
-Keep track of the effective copy_to list in the parser as an additional param to its methods, which will be cumbersome but doable
- Create a copy and change the underlying mappings copy_to, which is probably overkill

Is there another option that I'm not seeing here?

> Shouldn't we just not have a copy_to definition that targets a field that gets populated by this mapper to begin with?

Having copy_to targeting a semantic_text field is a valid use case - it's just that we don't want the parser to do the parsing, as the inference process will have already populated the inference results using all sources, so  the semantic text field mapper must not be invoked separately from the source fields.

Reply:
After talking offline with @original-brownbear , we're parking this for now and will optimize later if needed 👍